### PR TITLE
177 Onboarding Scope 3 indicator

### PR DESCRIPTION
**Pull request by drelsabrouty**
scope 3 indicator with assumed structure.
also still using the recursive limit override.

### CODE REVIEW

File: functions/processing_enriched_datamodel.py [hunk 1]
```diff
@@ -409,6 +410,81 @@ def generate_table(table_name: str) -> None:
...
+
+        ## PREPPING
+        covered_scope_3_output_ledger = ledger_ecoinvent_mapping.custom_join(covered_scope_3_output, custom_on="activity_uuid_product_uuid", custom_how="inner").custom_select(["tiltledger_id", "total_scope_3_emission_per_activity_uuid_product_uuid"])
+
+        ## CALCULATION
+        windowSpec = Window.partitionBy("tiltledger_id")
+        covered_scope_3_output_ledger.data = covered_scope_3_output_ledger.data.withColumn("total_scope_3_emission_per_ledger_id", F.sum("total_scope_3_emission_per_activity_uuid_product_uuid").over(windowSpec))
+        covered_scope_3_output_ledger = covered_scope_3_output_ledger.custom_drop(["total_scope_3_emission_per_activity_uuid_product_uuid"])
+        covered_scope_3_output_ledger = covered_scope_3_output_ledger.custom_select(["tiltledger_id", "total_scope_3_emission_per_ledger_id"]).custom_distinct()
```
Comment (SKruthoff) [inline functions/processing_enriched_datamodel.py:475]: I would renamne this column. I think this is not the total scope 3, so we should not call it like that. Additionally I think the names of the scope 1,2 and 3 tables should not contain the suffix of _enriched or _ledger

File: functions/processing_enriched_datamodel.py
Comment (SKruthoff) [inline functions/processing_enriched_datamodel.py:1]: The code works, but I think we are doing some redundant things like using the co2 data of the actual products which are not relevant. I am fine with leaving it like this to be honest.

**Review by SKruthoff — CHANGES_REQUESTED** [review]

**Comment by SKruthoff** [comment]
Lets fix the small things in this PR and then work on adding the custom groupby and changing some of the schema in the PRs to develop.

File: functions/processing_enriched_datamodel.py (hunk 1 above)
Comment (drelsabrouty) [reply functions/processing_enriched_datamodel.py]: Yes fixed, will do this for the other two tables of this immediately as part of commits towards the integrate_refactoring branch

File: functions/processing_enriched_datamodel.py
Comment (drelsabrouty) [reply functions/processing_enriched_datamodel.py:1]: agreed, lets do this as part of the integrate refactoring branch without needing to create separate prs for it (as the changes wont be major)

**Review by drelsabrouty — COMMENTED** [review]
checked

**Pull request merged**